File: KYCyberguard_backend.py
# KYCyberGuard Flask Backend

from flask import Flask, request, jsonify, render_template

app = Flask(__name__)

# Sample data to store assessments (in a real system, use a database)
assessments = []

@app.route('/')
def home():
    return render_template('index.html')

@app.route('/api/assess', methods=['POST'])
def assess():
    try:
        data = request.get_json()
        if not data or not isinstance(data, dict):
            raise ValueError("Invalid input data")
        
        company_name = data.get('company_name')
        cybersecurity_score = calculate_score(data)
        
        assessment = {
            "company_name": company_name,
            "score": cybersecurity_score,
            "recommendations": generate_recommendations(cybersecurity_score)
        }
        assessments.append(assessment)
        return jsonify({"status": "success", "assessment": assessment}), 200
    except ValueError as ve:
        return jsonify({"status": "error", "message": str(ve)}), 400
    except Exception as e:
        return jsonify({"status": "error", "message": "An unexpected error occurred"}), 500


def calculate_score(data):
    score = 0
    if data.get('has_antivirus'):
        score += 20
    if data.get('has_firewall'):
        score += 20
    if data.get('has_ids'):
        score += 20
    if data.get('employee_training'):
        score += 20
    if data.get('data_backup'):
        score += 20
    return score


def generate_recommendations(score):
    if score < 60:
        return [
            "Implement antivirus software",
            "Set up a firewall",
            "Conduct regular employee cybersecurity training"
        ]
    elif score < 80:
        return ["Improve data backup policies", "Enhance intrusion detection systems"]
    else:
        return ["Maintain current cybersecurity practices"]

if __name__ == '__main__':
    app.run(debug=False, use_reloader=False)

File: KYCyberguard_backend.js
// Import necessary modules
const express = require('express');
const cors = require('cors');

// Initialize the Express application
const app = express();
const PORT = 3000;

// Middleware
app.use(cors());
app.use(express.json());

// Function to calculate assessment score and recommendations
function assessCompany(data) {
    let score = 0;
    const recommendations = [];

    // Scoring criteria
    if (data.has_antivirus) {
        score += 20;
    } else {
        recommendations.push("Install antivirus software to protect against malware.");
    }

    if (data.has_firewall) {
        score += 20;
    } else {
        recommendations.push("Implement a firewall to monitor and control network traffic.");
    }

    if (data.has_ids) {
        score += 20;
    } else {
        recommendations.push("Set up an Intrusion Detection System (IDS) to detect potential threats.");
    }

    if (data.employee_training) {
        score += 20;
    } else {
        recommendations.push("Conduct regular cybersecurity training for employees.");
    }

    if (data.data_backup) {
        score += 20;
    } else {
        recommendations.push("Establish a regular data backup strategy.");
    }

    return { score, recommendations };
}

// API route
app.post('/api/assess', (req, res) => {
    const { company_name, has_antivirus, has_firewall, has_ids, employee_training, data_backup } = req.body;

    if (!company_name || typeof company_name !== 'string') {
        return res.status(400).json({ message: "Invalid or missing 'company_name'." });
    }

    const assessment = assessCompany({
        has_antivirus,
        has_firewall,
        has_ids,
        employee_training,
        data_backup
    });

    return res.json({
        assessment: {
            company_name,
            score: assessment.score,
            recommendations: assessment.recommendations
        }
    });
});

// Start the server
app.listen(PORT, () => {
    console.log(`Server is running on http://localhost:${PORT}`);
});

